File: server/src/main/java/com/mccorby/photolabeller/server/FederatedServer.java
package com.mccorby.photolabeller.server;

import com.mccorby.photolabeller.server.core.domain.model.*;
import com.mccorby.photolabeller.server.core.domain.repository.ServerRepository;
import org.nd4j.linalg.api.ops.impl.transforms.Round;

import java.io.File;
import java.io.InputStream;
import java.util.Properties;

public interface FederatedServer {

    void initialise(ServerRepository repository,
                    GradientStrategy gradientStrategy,
                    RoundController roundController,
                    Logger logger,
                    Properties properties);

    void pushGradient(InputStream clientGradient, int samples);

    UpdatingRound getUpdatingRound();

    File getModelFile();

    String getUpdatingRoundAsJson();
}

File: server/src/main/kotlin/com/mccorby/photolabeller/server/core/domain/model/FederatedServer.kt
package com.mccorby.photolabeller.server.core.domain.model

import com.mccorby.photolabeller.server.Logger
import com.mccorby.photolabeller.server.core.domain.repository.ServerRepository
import java.io.File
import java.io.InputStream
import java.util.*

interface FederatedServer {
    fun initialise(repository: ServerRepository,
                            gradientStrategy: GradientStrategy,
                            roundController: RoundController,
                            logger: Logger,
                            properties: Properties)

    fun pushGradient(clientGradient: InputStream, samples: Int)

    fun getUpdatingRound(): UpdatingRound

    fun getModelFile(): File

    fun getUpdatingRoundAsJson(): String
}